src/otx/core/data/entity/base.py

```diff
@@ -669,25 +669,29 @@ class OTXBatchPredEntity(OTXBatchDataEntity):
 
     Attributes:
         scores: List of probability scores representing model predictions.
-        saliency_maps: List of saliency maps used to explain model predictions.
+        saliency_map: List of saliency maps used to explain model predictions.
             This field is optional and will be an empty list for non-XAI pipelines.
-        feature_vectors: List of intermediate feature vectors used for model predictions.
+        feature_vector: List of intermediate feature vectors used for model predictions.
             This field is optional and will be an empty list for non-XAI pipelines.
     """
 
     scores: list[np.ndarray] | list[Tensor]
 
     # (Optional) XAI-related outputs
-    saliency_maps: list[np.ndarray] | list[Tensor] = field(default_factory=list)
-    feature_vectors: list[np.ndarray] | list[Tensor] = field(default_factory=list)
+    # TODO(vinnamkim): These are actually plural, but their namings are not
```

Comment:
`These are actually plural, but their namings are not`, this is what might lead to some minor confusion. I appreciate unification that you bring. Another good alternative was to instead update Model API to `saliency_maps` and `feature_vectors`.

Reply:
Yes, as I understood from Vinnam's comment, the plan is to update Model API and then revert these changes `saliency_map`->`saliency_maps`, `feature_vector` -> `feature_vectors`